app/models/manager_refresh/inventory_collection/builder/infra_manager.rb

```diff
@@ -173,13 +178,15 @@ def host_switches
 
         def switches
           add_properties(
-            :manager_ref => %i(uid_ems)
+            :manager_ref => %i(uid_ems), # TODO looks like switches are using a bad association, this one is defined as through hosts
```

Comment:
so this looks like we load switches only through hosts? And I am not sure if we were filling up the :ems_id. Also, the manager has_many :switches is missing dependent destroy, so those will be probably orphaned on provider delete

Reply:
Yeah the lack of a dependent destroy is because some switches are strictly per-host and some are shared.  There is a pretty gross "remove_obsolete_switches" in `EmsRefresh.save_inventory_infra` that does the cleanup.

What I want to do is move the shared switches up to belonging to the EMS and filling in the ems_id, and the host switches are under the host.  This way we can use the associations properly instead of this mess.

I got hung up on doing the data migration of existing switches but maybe we can tackle it now that we have more time.  Moving lans up to the EMS would be another good step since right now they're duplicated per switch.

Reply:
Right, so that means the current graph refresh will be leaving orphans. So we should either fix the ems_id, or somehow bring the custom deleting?